packages/access-token-jwt/test/dpop-verifier.test.ts

```diff
@@ -0,0 +1,1108 @@
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Missing "htm" in DPoP proof');
+  });
+
+  it('throws InvalidProofError if htm is not a string in DPoP proof', async () => {
+    const { publicKey, privateKey } = await generateKeyPair('ES256');
+
+    const jwt = 'header.payload.signature';
+    const encodedAth = crypto
+      .createHash('sha256')
+      .update(jwt)
+      .digest('base64url');
+
+    const proof = await new SignJWT({
+      htm: 12345 as any, // invalid type
+      htu: 'https://api.example.com/resource',
+      jti: crypto.randomUUID(),
+      iat: Math.floor(Date.now() / 1000),
+      ath: encodedAth,
+    })
+      .setProtectedHeader({
+        alg: 'ES256',
+        typ: 'dpop+jwt',
+        jwk: await exportJWK(publicKey),
+      })
+      .sign(privateKey);
+
+    const result = verifyDPoP({
+      jwt,
+      method: 'POST',
+      url: 'https://api.example.com/resource',
+      headers: {
+        authorization: `DPoP ${jwt}`,
+        dpop: proof,
+      },
+      supportedAlgorithms: ['ES256'],
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Invalid "htm" claim');
+  });
+
+  it('throws InvalidProofError if htu is missing in DPoP proof', async () => {
+    const { publicKey, privateKey } = await generateKeyPair('ES256');
+
+    const jwt = 'header.payload.signature';
+    const encodedAth = crypto
+      .createHash('sha256')
+      .update(jwt)
+      .digest('base64url');
+
+    const proof = await new SignJWT({
+      htm: 'POST',
+      // htu is intentionally omitted
+      jti: crypto.randomUUID(),
+      iat: Math.floor(Date.now() / 1000),
+      ath: encodedAth,
+    })
+      .setProtectedHeader({
+        alg: 'ES256',
+        typ: 'dpop+jwt',
+        jwk: await exportJWK(publicKey),
+      })
+      .sign(privateKey);
+
+    const result = verifyDPoP({
+      jwt,
+      method: 'POST',
+      url: 'https://api.example.com/resource',
+      headers: {
+        authorization: `DPoP ${jwt}`,
+        dpop: proof,
+      },
+      supportedAlgorithms: ['ES256'],
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Missing "htu" in DPoP proof');
+  });
+
+  it('throws InvalidProofError if "htu" claim is not a string', async () => {
```

Comment:
Isn't this testing the same thing as the test case called `throws if htu is not a string`?